### PR TITLE
build: add support for execution role deployment

### PR DESCRIPTION
**Pull request by ericwestfall**
This change introduces a new `deploy` top-level project directory that contains parameterized CloudFormation templates that can be used to deploy the required Starfleet execution roles.

The following CloudFormation templates are included:

The `starfleet-execution-role-stackset.yml` template supports deployment through CloudFormation StackSets and allows users to target deployment to all member accounts in an AWS Organization or specific targets (e.g. OUs, accounts, etc.).

The `starfleet-execution-role-org-management.yml` template is intended to be deployed as a standard CloudFormation Stack in the organization management account. The template creates a role for the Starfleet Account Index Generator and a standard Starfleet Worker execution role.

Both templates enforce strict input checking for all parameters and implement least-privilege policies. Lastly, this change includes new `tox` environments to check CloudFormation templates for linting and security misconfigurations.

### CODE REVIEW

**Comment by mikegrima** [comment]
There seems to be a regression with:
```
boto3==1.26.127
botocore==1.29.127
```

That's very strange that it would break SQS...

**Comment by mikegrima** [comment]
I wonder if this is the regression:
https://github.com/boto/botocore/commit/2395012bd369f89b0bcfbbf830718866fc6ae166#diff-53c9ed27fbb58a2b8274cc8203f6dbbf167840984a94be3b55df322762f01b5bR941-R944

**Comment by mikegrima** [comment]
This is the issue: https://github.com/getmoto/moto/issues/6286

**Comment by mikegrima** [comment]
@ericwestfall in the meantime, let's pin to the older versions of `boto3` and `botocore` in the `requirements.txt` file:
```
boto3==1.26.126
botocore==1.29.126
```

File: deploy/starfleet-execution-role-org-management.yml [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    MinLength: 1
+    MaxLength: 32
+  Partition:
+    Type: String
+    Description: >-
+      (Optional) The AWS partition name in which Starfleet resources are being deployed.
+    Default: aws
+    AllowedValues:
+      - aws
+      - aws-cn
```
Comment (mikegrima) [inline deploy/starfleet-execution-role-org-management.yml:93]: One issue right now is that starfleet doesn't have any logic for handling cn or gov-cloud partitions.

Comment (ericwestfall) [reply deploy/starfleet-execution-role-org-management.yml]: Fair shout, fixed in 5c37a219b5939d4e16f41939d923cfc393198711.

**Comment by ericwestfall** [comment]
> @ericwestfall in the meantime, let's pin to the older versions of `boto3` and `botocore` in the `requirements.txt` file:
> 
> ```
> boto3==1.26.126
> botocore==1.29.126
> ```

The underlying change in `botocore` that caused this regression has been rolled back in boto/botocore#2931. Looks like we can safely pin to the latest versions at this stage.

**Comment by mikegrima** [comment]
LGTM. The next step will be to update the docs to reference how to use the CF templates for the role.

**Review by mikegrima — APPROVED** [review]

**Pull request merged**